tests/unit/BackendClient-spec.js

```diff
@@ -0,0 +1,183 @@
+        };
+        var sessionDurationTest = config.get('backendClient').ksSessionRefreshIntervalMinutes * 60 * 1000;
+        var fakeTime = sessionDurationTest + 1;
+        clock = sinon.useFakeTimers(fakeTime);
+        var BackendClient = createBackendClient(mocks);
+        BackendClient.SetLastSessionTime(0);
+        BackendClient.getLiveEntriesForMediaServer().then(
+            function () {
+                console.log(BackendClient.GetlastSessionTime());
+                console.log('Promise Succsed');
+                expect(BackendClient.GetlastSessionTime()).to.eql(fakeTime);
+                done();
+            },
+            function(err)
+            {
+                console.log('Promise failed');
+                done(err);
+            });
+    });
+    it ("Promise should be failed, after error occurred, in the create new session",function(done){
+        KalturaClientMock = {
+            KalturaClient : sinon.stub().returns({
+                session : {
+                    start : sinon.stub().callsArgWith(0,null,'1')
+                },
+                setSessionId: sinon.stub()
+            })
+        };
+        var mocks = {
+            './logger/logger': sinon.stub().returns(loggerMock),
+            './kaltura-client-lib/KalturaClient' : KalturaClientMock
+        };
+        var sessionDurationTest=config.get('backendClient').ksSessionRefreshIntervalMinutes* 60 * 1000;
+        var FakeTime=sessionDurationTest+1;
+        clock = sinon.useFakeTimers(FakeTime);
+        var BackendClient=createBackendClient(mocks);
+        BackendClient.SetLastSessionTime(0);
+        var promise = BackendClient.getLiveEntriesForMediaServer();
+
+        promise.then(function(){
+            done(new Error("Expected getLiveEntries to fail"));
+        }, function(){
+            done();
+        });
+    });
+    it("Calling the function .getLiveEntriesForMediaServer() twice, check that function create new session is called once!",function(done){
+        setSessionIdMock=sinon.spy();
+        KalturaClientMock = {
+            KalturaClient : sinon.stub().returns({
+                session : {
+                    start : sinon.stub().callsArgWith(0,'1')
+                },
+                setSessionId: setSessionIdMock,
+                liveStream: {
+                    listAction: sinon.stub().callsArgWith(0, '1')
+                }
+            })
+        };
+        var mocks = {
+            './logger/logger': sinon.stub().returns(loggerMock),
+            './kaltura-client-lib/KalturaClient' : KalturaClientMock
+        };
+        var FakeTime=1;
+        clock = sinon.useFakeTimers(FakeTime);
+        var BackendClient=createBackendClient(mocks);
+        expect(setSessionIdMock.callCount).to.eql(0);
+        BackendClient.getLiveEntriesForMediaServer()
+            .then(function(){
+                expect(setSessionIdMock.callCount).to.eql(1);
+                clock = sinon.useFakeTimers(FakeTime+1);
```

Comment:
if you know that fake time is 1, I think writing 2 here would be more readable :)